energy-integration/common-modules/weekly-rewards-splitting/src/base_impl.rs

```diff
@@ -0,0 +1,85 @@
+elrond_wasm::imports!();
+
+use common_types::{TokenAmountPairsVec};
+use week_timekeeping::Week;
+
+use crate::{events, ClaimProgress};
+
+pub trait AllBaseWeeklyRewardsSplittingImplTraits = crate::WeeklyRewardsSplittingModule
+    + energy_query::EnergyQueryModule
+    + week_timekeeping::WeekTimekeepingModule
+    + events::WeeklyRewardsSplittingEventsModule;
+
+pub trait WeeklyRewardsSplittingTraitsModule {
```

Comment:
To make it easier to implement on the farm side, also have `&self` in the methods. (making them actual methods instead of static functions for the type)

Reply:
This also means the methods using the wrapper, will have to have an argument of its type.

Reply:
Done.